commons/src/main/java/com/expedia/www/haystack/pipes/commons/kafka/config/FirehoseConfig.java

```diff
@@ -0,0 +1,63 @@
+package com.expedia.www.haystack.pipes.commons.kafka.config;
+
+public class FirehoseConfig {
+
+    private String url;
+
+    private String streamName;
+
+    private String signingRegion;
+
+    private int initialRetrySleep;
+
+    private int maxRetrySleep;
+
+    private boolean useStringBuffering;
+
+    private int maxBatchInterval;
+
+    private int maxParallelISMPerShard;
+
+    public FirehoseConfig(final String url, final String streamName, final String signingRegion, final int initialRetrySleep, final int maxRetrySleep, final boolean useStringBuffering, final int maxBatchInterval, final int maxParallelISMPerShard) {
```

Comment:
Would it be worth it to validate some of these configurations? if stuff is missing perhaps. Would be ideal to know right away instead of finding down the call stack that something was not setup correctly.

Reply:
I am taking care of it in ProjectConfiguration.class.